src/Identity.php

```diff
@@ -71,15 +72,37 @@ public function getIdentifier()
         return $this->get('id');
     }
 
+    /**
+     * Get data from the identity using object access.
+     *
+     * @param string $field Field in the user data.
+     * @return mixed
+     */
+    public function __get($field)
+    {
+        return $this->get($field);
+    }
+
+    /**
+     * Check if the field isset() using object access.
+     *
+     * @param string $field Field in the user data.
+     * @return mixed
+     */
+    public function __isset($field)
+    {
+        return $this->get($field) !== null;
+    }
+
     /**
      * Get data from the identity
      *
      * @param string $field Field in the user data.
      * @return mixed
      */
-    public function get($field)
+    protected function get($field)
```

Comment:
Oh please, no! Why??? I'd like to have the choice to use get over property access.

I prefer to use get and I really hate having protected methods without the _ prefix. The prefixing is really a helper to identify on sight if it's protected or not.

Reply:
So you want to have:

* `get(string)`
* `offsetGet()`
* `__get()`

All 3 methods don't allow typehinting, so I don't know why we'd allow all 3. Would you be open to dropping `ArrayAccess` and keeping `get()` and `__get()`?

Reply:
I’d stick with ArrayAccess since we force the underlying object to be array|ArrayAccess. This would allow to continue using this interface.

Reply:
I am also against this, as this is a code smell for unknown keys (compared to entities where the keys and their types are known): https://github.com/cakephp/authentication/issues/249

Reply:
That is not true, get() can be made typehinted dynamically via IDE extension, the others cannot.
See [meta](https://github.com/dereuromark/cakephp-ide-helper/blob/master/tests/test_files/meta/phpstorm/.meta.php) file.